swan-lake/learn-the-platform/test-document-the-code/test-ballerina-code/mocking.md

```diff
@@ -39,8 +39,37 @@ structurally equivalent to the real object via the mocking features in the test
 
 ***Example:***
 
-Let's make changes to the example in the [Test a simple function](/learn/test-ballerina-code/test-a-simple-function/) to define a 
-test double for the `clientEndpont` object.
+Consider the following example in which an `http:Client` interacts with an external endpoint to get a random joke.
+
+***main.bal***
+
+```ballerina
+import ballerina/io;
+import ballerina/http;
+import ballerina/regex;
+
+http:Client clientEndpoint = check new ("https://api.chucknorris.io/jokes/");
+
+// This function performs a `get` request to the Chuck Norris API and returns a random joke 
+// with the name replaced by the provided name or an error if the API invocation fails.
+function getRandomJoke(string name) returns string|error {
+
+    http:Response response = check clientEndpoint->get("/random");
+
+    if response.statusCode != http:STATUS_OK {
+        string errorMsg = "error occurred while sending GET request";
+        io:println(errorMsg, ", status code: ", response.statusCode, ", payload: ", response.getJsonPayload());
+        return error(errorMsg);
+    }
+
+    json payload = check response.getJsonPayload().ensureType();
```

Comment:
Shouldn't we simplify this using data binding? Do we really need the response/status codes? Even if we do, we may be able to check by `http:ApplicationResponseError`.

Reply:
Fixed with 539d304dea66d953a5d45925210ca0a017f58bdf